File: KuantumKaos/java/kuantumKaosJava/src/AntiMadde.java
public class AntiMadde extends KuantumNesnesi implements IKritik {

    @Override
    public void analizEt() {
        stabilite -= 35;
        System.out.println("Anti-madde tepkimeye giriyor!");

        if (stabilite <= 0) {
            throw new KuantumCokusuException(id);
        }
    }

    @Override
    public void acilDurumSogutmasi() {
        stabilite += 50;
        System.out.println("Anti-madde kararlılığı artırıldı.");
    }
}

File: KuantumKaos/javascript/js_kaos/AntiMadde.js
import { KuantumNesnesi } from "./KuantumNesnesi.js";
import { IKritik } from "./IKritik.js";
import { KuantumCokusuException } from "./KuantumCokusuException.js";

export class AntiMadde extends KuantumNesnesi {
    analizEt() {
        this.stabilite -= 35;
        console.log("Anti-madde tepkimeye giriyor!");

        if (this.stabilite <= 0) {
            throw new KuantumCokusuException(this.id);
        }
    }

    acilDurumSogutmasi() {
        this.stabilite += 50;
        console.log("Anti-madde kararlılığı artırıldı.");
    }
}
